test/compilable/issue21203.d

```diff
@@ -0,0 +1,208 @@
+    static assert(funi.mangleof == "_ZN2ns4funiERKNS_8MyClassBE");
+    static assert(funj.mangleof == "_ZN2ns4funjEPKNS_8MyClassBE");
+}
+else version (CppRuntime_Microsoft)
+{
+    static assert(funf.mangleof == "?funf@ns@@YAXVMyClassB@1@@Z");
+    static assert(fung.mangleof == "?fung@ns@@YAXVMyClassB@1@@Z");
+    static if (size_t.sizeof == ulong.sizeof)
+    {
+        static assert(funh.mangleof == "?funh@ns@@YAXAEAVMyClassB@1@@Z");
+        static assert(funi.mangleof == "?funi@ns@@YAXAEBVMyClassB@1@@Z");
+        static assert(funj.mangleof == "?funj@ns@@YAXPEBVMyClassB@1@@Z");
+    }
+    else
+    {
+        static assert(funh.mangleof == "?funh@ns@@YAXAAVMyClassB@1@@Z");
+        static assert(funi.mangleof == "?funi@ns@@YAXABVMyClassB@1@@Z");
+        static assert(funj.mangleof == "?funj@ns@@YAXPBVMyClassB@1@@Z");
+    }
+}
+
+//Templates
+extern(C++)
+{
+    void funTempl(T)();
+    class MyClassC {}
+    alias funTemplA = funTempl!(ScopeClass!MyClassC);
+    alias funTemplB = funTempl!(const ScopeClass!MyClassC);
+    alias funTemplC = funTempl!(const(ScopeClass!MyClassC)*);
+    // N.B funTempl!([const] ref ScopeClass!MyClassC) is not permissable in D
+}
+version (Posix)
+{
+    static assert(funTemplA.mangleof == "_Z8funTemplI8MyClassCEvv");
+    static assert(funTemplB.mangleof == "_Z8funTemplIK8MyClassCEvv");
+    static assert(funTemplC.mangleof == "_Z8funTemplIPK8MyClassCEvv");
+}
+else version (CppRuntime_Microsoft)
+{
+    static assert(funTemplA.mangleof == "??$funTempl@VMyClassC@@@@YAXXZ");
+    static assert(funTemplB.mangleof == "??$funTempl@$$CBVMyClassC@@@@YAXXZ");
+    static if (size_t.sizeof == ulong.sizeof)
+        static assert(funTemplC.mangleof == "??$funTempl@PEBVMyClassC@@@@YAXXZ");
+    else
+        static assert(funTemplC.mangleof == "??$funTempl@PBVMyClassC@@@@YAXXZ");
+}
+
+template _function(F)
+{
+extern(C++, "std")
+{
+    extern(C++, struct)
+    pragma(mangle, "function")
+    class _function
+    {
+    }
+}
+}
+template FunctionOf(F)
+{
+    F f;
+    alias FunctionOf = typeof(*f);
+}
+extern(C++) void funk(ScopeClass!(_function!(FunctionOf!(void function(int))),"function") a ){ }
+
+version (Posix)
+{
+    static assert(funk.mangleof == "_Z4funkSt8functionIFviEE");
+}
+else version (CppRuntime_Microsoft)
+{
+    static assert(funk.mangleof == "?funk@@YAXV?$function@$$A6AXH@Z@std@@@Z");
+}
+
+extern(C++, "ns")
+{
+    pragma(mangle, "function")
+    class _function2
+    {
+        public final void test();
+    }
+}
+
+version (Posix)
+{
+    static assert(_function2.test.mangleof == "_ZN2ns8function4testEv");
+}
+else version (CppRuntime_Microsoft)
+{
+    static if (size_t.sizeof == ulong.sizeof)
+        static assert(_function2.test.mangleof == "?test@function@ns@@QEAAXXZ");
+    else
+        static assert(_function2.test.mangleof == "?test@function@ns@@QAEXXZ");
+}
+
+extern(C++, "ns")
+{
+    template _function3(T)
+    {
+        pragma(mangle, _function3, "function")
```

Comment:
This doesn't seem to match what the corresponding Spec pull says about 3 arguments.

Reply:
Spec PR will be update to match this once it goes in.